test/vtquery.test.js

```diff
@@ -681,11 +681,149 @@ test('options - dedupe: compare fields from real-world tiles (increases coverage
+  const opts = {
+    radius: 800, // about the width of a z15 tile
+    filters: [{'parameter': 'bool_value', 'condition': '!=', 'value': false}]
+  };
+  vtquery(tiles, [-122.3384, 47.6635], opts, function(err, result) {
+    assert.equal(result.features.length, 1, 'expected one feature');
+    const p = result.features[0].properties;
+    assert.equal(p.bool_value, true, 'expected value type');
+    assert.ifError(err);
+    assert.end();
+  });
+});
+
+test('options - filter: Test Float Approximate Equals', assert => {
+  const tiles = [{buffer: mvtf.get('038').buffer, z: 15, x: 5248, y: 11436}];
+  const opts = {
+    radius: 800, // about the width of a z15 tile
+    filters: [{'parameter': 'float_value', 'condition': '=', 'value': 3.1}]
+  };
+  vtquery(tiles, [-122.3384, 47.6635], opts, function(err, result) {
+    assert.equal(result.features.length, 1, 'expected one feature');
+    const p = result.features[0].properties;
+    assert.equal(p.float_value, 3.0999999046325684, 'expected value type');
+    assert.ifError(err);
+    assert.end();
+  });
+});
+
+test('options - filter: Test No Results Less Than Equals', assert => {
+  const tiles = [{buffer: mvtf.get('038').buffer, z: 15, x: 5248, y: 11436}];
+  const opts = {
+    radius: 800, // about the width of a z15 tile
+    filters: [{'parameter': 'uint_value', 'condition': '<', 'value': 0}]
+  };
+  vtquery(tiles, [-122.3384, 47.6635], opts, function(err, result) {
+    assert.equal(result.features.length, 0, 'expected one feature');
+    assert.ifError(err);
+    assert.end();
+  });
+});
+
+test('options - filter: Test No Results Greater Than Equals', assert => {
+  const tiles = [{buffer: mvtf.get('038').buffer, z: 15, x: 5248, y: 11436}];
+  const opts = {
+    radius: 800, // about the width of a z15 tile
+    filters: [{'parameter': 'sint_value', 'condition': '>', 'value': 0}]
+  };
+  vtquery(tiles, [-122.3384, 47.6635], opts, function(err, result) {
+    assert.equal(result.features.length, 0, 'expected one feature');
+    assert.ifError(err);
+    assert.end();
+  });
+});
+
+test('options - filter: Test Double Int Greater Than Equal', assert => {
+  const tiles = [{buffer: mvtf.get('038').buffer, z: 15, x: 5248, y: 11436}];
+  const opts = {
+    radius: 800, // about the width of a z15 tile
+    filters: [{'parameter': 'double_value', 'condition': '>', 'value': -2}]
+  };
+  vtquery(tiles, [-122.3384, 47.6635], opts, function(err, result) {
+    assert.equal(result.features.length, 1, 'expected one feature');
+    const p = result.features[0].properties;
+    assert.equal(p.double_value, 1.23, 'expected value type');
+    assert.ifError(err);
+    assert.end();
+  });
+});
+
 test('success: returns all possible data value types', assert => {
   const tiles = [{buffer: mvtf.get('038').buffer, z: 15, x: 5248, y: 11436}];
   const opts = {
     radius: 800 // about the width of a z15 tile
-  }
+  };
```

Comment:
Thanks for adding tests! I think I'd feel a little more comfortable adding some extra tests with some additional fixtures (other than the mvt-fixtures038 tile) to ensure we're properly returning _some_ values, but not others, based on the filter. Right now I'm not confident the filter is working since the 038 tile only has a single feature in it. 

I can work on a new mvt-fixture that has multiple points with different property values that will help in constructing more holistic tests!

Reply:
That would be great!  Yeah, I don't have a great idea on how to get data/make data yet so I was just working with what I found.